### ISSUE TITLE
VSCode Extension - Getting started - Download Sample App - Unsupported response type

### ISSUE THREAD

**Issue by bolt-io** [microsoft/vscode-ms-entra#10]
The sample app setup does not set the response type to 'code' and thus throws an exception upon redirecting back.

Sample: ASP.NET Core

![image](https://github.com/microsoft/vscode-ms-entra/assets/10209663/84fc332e-01b4-4e88-a30d-e3f607b86c19)

**Comment by bktowett** [comment]
Hi @bolt-io Acknowledging this.

**Comment by emakuba** [comment]
Hello @bolt-io, for ASP.NET it's a requirement to select ID tokens (under implicit grants and hybrid flows) for the app requesting tokens. Something that our VS Code extension doesn't do by default while registering the app on Azure. To unblock you (quick fix), please go to Azure click the app and select ID tokens in the Redirect URIs section. I will give this feedback to my team, to deliberate whether we can auto-select ID tokens on app registration for web apps (as a long-term solution).

**Comment by bolt-io** [comment]
Hi @emakuba, thanks for your response.

We know there's a whole plethora of security concerns with implicit grant, can the development team not just use the auth code flow with PKCE instead? It is supported with the packages the sample uses.

It's not very good for Microsoft samples to be using outdated auth flows that are out of scope for the next OAuth 2.1 iteration 

![image](https://github.com/microsoft/vscode-ms-entra/assets/10209663/70717a4a-a7d3-460b-952f-2fc1019b8a8b)

**Comment by emakuba** [comment]
Thanks @bolt-io for your suggestion, I will share it with my team for consideration.

**Comment by emakuba** [comment]
@bolt-io I notified the sample team maintaining the ASP.NET core sample, they are working on a fix to support a more secure grant type (as a long-term solution). We made a transient fix to enable id token endpoint when downloading this sample, so users are now able to successfully sign-in after downloading the app.
